### PR TITLE
Table panel: Make filter case insensitive

### LINKED ISSUE #39283: Column filter not case-insensitive
**What happened**:
I activated column filtering and started typing "cross" in the filter field, and no result was found

**What you expected to happen**:
I expected it to find:
- Cross-group next generation middleware
- Cross-platform bottom-line circuit

**How to reproduce it (as minimally and precisely as possible)**:
1. Open the [playground (link to specific panel edit)](https://play.grafana.org/d/OhR1ID6Mk/3-table?orgId=1&editPanel=12) and enable "Column filter"
2. Click on the filter icon next to the column header "Catch phrase"
3. Type "cross"

image

**Anything else we need to know?**:
I created this as a bug report since the similar issue #30476 was also flagged as Bug

**Environment**:
- Grafana version: 7.5
- Data source type & version: SQL (but doesn't really matter)
- OS Grafana is installed on: Debian
- User OS & Browser: Win 10, Firefox or Brave

### PR DESCRIPTION
**Pull request by dprokop**
Fixes https://github.com/grafana/grafana/issues/39283

Instead of using Input field I've change it to FilterInput that we use in similar filtering scenarios. 

FilterInput & Input changes cherry-picked from https://github.com/grafana/grafana/pull/39073

### CODE REVIEW

**Review by connorlindsey — APPROVED** [review]
LGTM

File: packages/grafana-ui/src/components/Table/FilterList.tsx [hunk 1]
```diff
@@ -19,16 +19,16 @@ export const FilterList: FC<Props> = ({ options, values, onChange }) => {
   const theme = useTheme2();
   const styles = getStyles(theme);
   const [searchFilter, setSearchFilter] = useState('');
-  const items = useMemo(() => options.filter((option) => option.label?.indexOf(searchFilter) !== -1), [
-    options,
-    searchFilter,
-  ]);
+  const items = useMemo(
+    () => options.filter((option) => option.label?.toLowerCase().indexOf(searchFilter.toLowerCase()) !== -1),
```
Comment (hugohaggmark) [inline packages/grafana-ui/src/components/Table/FilterList.tsx:23]: nit: maybe change to `includes` instead
```suggestion
    () => options.filter((option) => option.label?.toLowerCase().includes(searchFilter.toLowerCase())),
```

**Review by hugohaggmark — APPROVED** [review]
Looks great and works as expected 🎉

**Comment by hugohaggmark** [comment]
> Thanks for the fix. How will I know when this is in a release? Just checking the release notes?

Hi @jtommi this should already be in 8.2.0-beta2

**Comment by dprokop** [comment]
@jtommi ah sorry - my bad for not adding this to changelog :)

**Comment by jtommi** [comment]
@dprokop I realize that special characters like period ( . )or dash ( - ) are not searchable.
I have no idea if they were before and as a consequence whether it's a bug created by this PR or not.
Should I just create a new issue?
Go [here](https://play.grafana.org/d/OhR1ID6Mk/3-table?orgId=1&editPanel=12), active column filter and search for "cross-"

**Comment by jtommi** [comment]
@dprokop  Great work, thanks

**Pull request merged**